_posts/2016-05-23-opaque-tokens-in-angular-2.md

```diff
@@ -21,13 +21,13 @@ author: pascal_precht
 
 If you've read our article series on everything dependency injection in Angular 2, you've probably realised that Angular is doing a pretty good job on that. We can either use string or type tokens to make dependencies available to the injector. However, when using string tokens, there's a possibility of running into naming collisions because... well, maybe someone else has used the same token for a different provider. In this article we're going to learn how so called "opaque tokens" solve this problem.
 
-Let's first recap what the differents between a string token and a type token is, before we jump into the actual problem we want to solve.
+Before we jump into the actual problem we want to solve, let's first recap the differences between a string token and a type token.
 
 ## String Tokens vs Type Tokens
 
-In order to associate a dependency creation with a token that we use throughout our application, we have to setup providers. A couple weeks ago we've written about how [providers can be created using Map literals](/angular/2016/05/13/angular-2-providers-using-map-literals.html), if you haven't read this one yet we recommend to check it out, as this article pretty much builds up on that one.
+Angular 2 uses dependency factories to create singleton instances that will be used in the runtime DI processes. In order to associate a token with a dependency factory, we have to setup providers. A couple weeks ago we blogged how [providers can be created using Map literals](/angular/2016/05/13/angular-2-providers-using-map-literals.html), if you haven't read this one yet we recommend to check it out, as this article pretty much builds up on that one.
```

Comment:
The important aspect here is that "DI injects singleton instances which are created by provider-registered factories."